nautobot/ipam/models.py

```diff
@@ -1181,6 +1171,19 @@ def siblings(self, include_self=False):
 
         return query
 
+    @property
+    def _namespace(self):
+        # if a namespace was explicitly set, use it
+        if getattr(self, "_provided_namespace", None):
+            return self._provided_namespace
+        if self.parent is not None:
+            return self.parent.namespace
+        return get_default_namespace()
```

Comment:
This also "feels" off to me - so if we have an IPAddress with no parent, it's always going to be implicitly in the global namespace?

Reply:
If you try to create an IPAddress without specifying a namespace or parent (in the ORM only) it will try to find a valid parent in the default namespace and if one can be found it assigns that parent. This isn't changing that behavior, just moving it out of the model `__init__` so that creating a new instance doesn't hit the database.

The IPAddress namespace/parent logic is a non-standard workflow. In the UI if you're creating an IPAddress you can only select the namespace but we use that namespace to try to find a valid parent in that namespace and if we don't find one we fail validation in `clean`/`save`.